src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/settings/ManagedIndexSettings.kt

```diff
@@ -45,6 +46,18 @@ class ManagedIndexSettings {
             Setting.Property.Dynamic
         )
 
+        // 0: enabled, use onMaster time as ISM template last_updated_time
+        // -1: migration ended successfully
+        // -2: migration ended unsuccessfully
+        // >0: use this setting (epoch millis) as ISM template last_updated_time
+        val TEMPLATE_MIGRATION_CONTROL: Setting<Long> = Setting.longSetting(
```

Comment:
From what I've seen this setting is doing multiple things:
1. Determining whether or not template migration is enabled
2. Configuring migration behavior (like what template last_updated_time is used)
3. Propagating migration success/failure status

Since this setting is user configurable as well, is this kind of usage conventional or are we overloading the use of it?

Reply:
The idea is:
User can set it to `0` value to do migration; set it to `-1` to stop migration
After starting the migration, the process is quick, like minutes after, they can check if it goes well from the value `-1/-2`

If the value set to a positive value, it will be used as the take effect time for ISM template
for example: 1 -> `Thursday, January 1, 1970 12:00:01 AM`
ISM template will auto manage all index created after this time, if the index matches the index pattern in the ISM template.

I think this should be a fairly straight forward experience from user's point of view